src/game/Utils/Event_Pump.cc

```diff
@@ -45,33 +45,15 @@ static void AddEvent(GAMEEVENT const& gameEvent, UINT16 const usDelay, EventQueu
 	GetQueue(ubQueueID).push_back(pEvent);
 }
 
-
-static EVENT* RemoveEvent(UINT32 uiIndex, EventQueueID ubQueueID)
-try
+// Remove the first element from a queue and return it.
+// Calling this function on an empty queue is undefined behavior, don't do it.
```

Comment:
Why not assert it?

Reply:
I'd like to reverse the question: why should we put an assertion here? From a code review POV this function has only two callsites, both in the same file, so it's easy enough to verify they're ok. Why burden correct code with extra checks when I've just removed unnecessary null checks in this patch?

So an assert would only protect against someone calling this function incorrectly sometime in the future. But if you don't trust your contributors to call a simple four line function with one parameter correctly you'd have to place asserts in every function and forbid the use of the entire standard library because these functions don't babysit their users either.

And finally, asserts are ugly. IMO this code base has too much debug messages, not too few. If you compare the current master against the original JA2 code release you'll notice that many of the debug messages have been there for more than twenty years. I've even seen one instance (in Rotting_Corpses.cc) where the debug messages were never in a JA2 release, only in the JA2 beta/test versions, but they still live on in Stracciatella. I hope these messages helped fix whatever problem they were supposed to find, but if not then maybe it's time to admit they've failed to do their job? In both cases it would make sense to remove them.

Reply:
Because assumptions are just asking for trouble. Someone adds a new user down the line or changes the internals and things can go south. It doesn't hurt to be explicit, but neither did I say we should be 100% defensively programming.

An assert is a single line of code ensuring consistency in debug builds and has nothing to do with logging. But yeah, debug calls can surely be cleaned up, and if we don't expose the log levels to users yet, that's a simple way to add filtering.